File: analyzer-smallseg/python/benchmark.py
#encoding=utf-8
try:
    import psyco
    psyco.full()
except:
    pass

s3 = file("text.txt").read()
words = [x.rstrip() for x in file("main.dic") ]
from smallseg import SEG
seg = SEG()
print 'Load dict...'
seg.set(words)
print "Dict is OK."
from time import time

for i in xrange(1,101):
    start = time()
    for j in xrange(0,i):
        A = seg.cut(s3)
    cost = time()-start
    print i,"times, cost:",cost

print "********************************"

File: analyzer-smallseg/src/main/java/cc/pp/analyzer/smallseg/examples/Benchmark.java
package cc.pp.analyzer.smallseg.examples;

import java.io.File;
import java.io.IOException;

import org.apache.commons.io.FileUtils;

import cc.pp.analyzer.smallseg.core.Seg;

public class Benchmark {

	public static void main(String[] args) throws IOException {

		Seg seg = new Seg();
		seg.useDefaultDict();
		File textFile = new File(Thread.currentThread().getContextClassLoader().getResource("text.txt").getPath()
				.replace("%20", " "));
		String text = FileUtils.readFileToString(textFile,"utf-8");
		for (int i = 1; i < 101; i++) {
			long start = System.currentTimeMillis();
			for (int j = 0; j < i; j++)
				seg.cut(text);
			long cost = System.currentTimeMillis() - start;
			System.out.println(i + "times,cost:" + cost);
		}
	}
}
